tools/istio-iptables/pkg/builder/iptables_builder.go

```diff
@@ -40,6 +40,10 @@ type IptablesConsumer interface {
 
 // IptablesBuilder is a higher level interface based on builder pattern.
 type IptablesBuilder interface {
+	// NewChainV4 creates new IPv4 chain
+	NewChainV4(chain string, table string) IptablesBuilder
```

Comment:
I don't understand why you'd need those methods.
Can't you infer the list of chains to create from the rules? If a rule refers to a chain, that chain implicitly needs to be created.

Reply:
You can exclude standard chains by having a constant set of standard chain names. That way you can avoid trying to recreate a standard chain.

Reply:
Another advantage is that you can then always create all the chains first, before any rule insertion.
IMHO that would be cleaner, and consistent with what iptables-restore does.

Reply:
For now, I have added a `TODO`. I will fix it in next PR

Reply:
There are some `ip` commands that need to be taken out as well.

Reply:
I would prefer to eliminate the `NewChain*` methods now. I don't think this should be much effort.

Reply:
will remove NewChain* and fix Build*()

Reply:
You can remove these methods now that nothing is calling them anymore.